sourcegen-model/src/main/java/io/micronaut/sourcegen/model/TypeDef.java

```diff
@@ -448,4 +470,34 @@ public boolean isArray() {
             return true;
         }
     }
+
+    /**
+     * A combined type for representing a TypeDef with annotations.
+     *
+     * @param typeDef       The raw type definition
+     * @param annotations   List of annotations to associate
+     * @author Elif Kurtay
+     * @since 1.4
+     */
+    @Experimental
+    record AnnotatedTypeDef(TypeDef typeDef, List<AnnotationDef> annotations)  implements TypeDef {
```

Comment:
Turn `AnnotatedTypeDef` into an interface so `annotated` can be overridden by class type def